localstack/utils/aws/aws_stack.py

```diff
@@ -98,7 +98,7 @@ def get_s3_hostname():
 
 
 def fix_account_id_in_arns(
-    response, replacement: str, colon_delimiter: str = ":", existing: str | list[str] = None
+    response, replacement: str, colon_delimiter: str = ":", existing: str | List[str] = None
```

Comment:
The type operator for union types was introduced with 3.10, so this won't fix the issue I'm afraid...

Reply:
The CLI Tests / cli-tests (3.8) job seems to be passing but I've gone ahead and changed this in 5f755f268